src/components/datetime/QDatetime.js

```diff
@@ -26,25 +26,35 @@ export default {
   mixins: [FrameMixin],
   props: extend(
     input,
-    inline
+    inline,
+    {
+      popover: Boolean,
+      modal: Boolean
+    }
   ),
   data () {
-    let data = this.isPopover() ? {} : {
+    let data = this.isPopover ? {} : {
       transition: __THEME__ === 'ios' ? 'q-modal-bottom' : 'q-modal'
     }
     data.focused = false
-    data.model = clone(this.value === 0 || this.value ? this.value : this.defaultSelection)
+    data.model = clone(isValid(this.value) ? this.value : this.defaultSelection)
+    data.platformReactive = this.$q.platform // Make reactive. These values will probably never change, but just in case
```

Comment:
let's remove platformReactive and directly use $q.platform.
This is something that definitely won't change.